runtime/src/main/java/org/corfudb/runtime/view/ObjectBuilder.java

```diff
@@ -116,6 +117,20 @@ public T open() {
                         getCorfuSMRProxy())
                         .getSerializer();
 
+                // FIXME: temporary hack until we have a registry
+                // If current map in cache has no indexer, or there is currently an other one,
+                // this will create and compute the indices.
+                if (result instanceof CorfuTable) {
+                    CorfuTable currentCorfuTable = ((CorfuTable) result);
+                    if (arguments.length > 0) {
+                        // If current map in cache has no indexer, or there is currently an other index
+                        if (!(currentCorfuTable.hasIndices()) ||
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)